### DISCUSSION TITLE
OpenAPI points errors when Enum have just one value

### DISCUSSION THREAD

**Discussion by dmelo** [fastapi/fastapi#9990] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
import enum
from openapi3 import OpenAPI
from fastapi.openapi.utils import get_openapi

app = FastAPI()

class TestEnum(enum.Enum):
    ONE = "ONE"
    # TWO = "TWO" # <-- It works if the enum have more than 1 value

@app.get("/items/{item_id}")
def read_item(item_id: int, q: TestEnum | None = None):
    return {"item_id": item_id, "q": q}

openapi_schema = get_openapi(
        title="Testing",
        version="0.1.0",
        routes=app.routes,
)
openapi = OpenAPI(openapi_schema, validate=True)
assert not openapi.errors(), openapi.errors()
```

### Description

Running this code `pipenv run python main.py` will generate the error:

```
Traceback (most recent call last):
  File "/home/dmelo/proj3/fc/test_enum_one_value/main.py", line 22, in <module>
    assert not openapi.errors(), openapi.errors()
AssertionError: [SpecError("Expected components.schemas.TestEnum to be of type Schema or Reference, but did not find required fields [] or '$ref'"), ReferenceResolutionError("Invalid path ['components', 'schemas', 'HTTPValidationError'] in Reference")]
```

If you uncomment the item on the enum `TWO = "TWO"` and run the same command, then it works.

It looks like it fails to process the enum schema, if it only have one value.

### Operating System

Linux

### Operating System Details

```
Linux solomon 6.3.6-100.fc37.x86_64 #1 SMP PREEMPT_DYNAMIC Mon Jun  5 15:44:30 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

### FastAPI Version

0.100.1

### Python Version

3.11.4

### Additional Context

```
❯ pip freeze
annotated-types==0.5.0
anyio==3.7.1
certifi==2023.7.22
charset-normalizer==3.2.0
fastapi==0.100.1
idna==3.4
openapi3==1.8.1
pydantic==2.1.1
pydantic_core==2.4.0
PyYAML==6.0.1
requests==2.31.0
sniffio==1.3.0
starlette==0.27.0
typing_extensions==4.7.1
urllib3==2.0.4
```

**Comment by arjwilliams** [comment, marked as answer]
This is an issue with the `openapi3` library, not FastAPI/Pydantic.

The only difference between the two versions of the generated `openapi_schema` is at `components.schemas.TestEnum`:

With the single value in the enum, the JSON schema of TestEnum is:
```json
{"const": "ONE", "title": "TestEnum"}
```
With the two values in the enum, the JSON schema of TestEnum is:
```json
{"type": "string", "enum": ["ONE", "TWO"], "title": "TestEnum"}
```
Both of these are valid JSON schemas (you can try these out at https://www.jsonschemavalidator.net/ or https://jsonschema.dev/), so therefore the issue is with the `openapi3` library having a false negative when validating the schema.

@dmelo You can raise this issue in the `openapi3` repo, and/or if you still need an OpenAPI spec validation library, and are willing to switch, then one that seems to work for this case is https://pypi.org/project/openapi-spec-validator/:

```python
from fastapi import FastAPI
import enum
from openapi_spec_validator import validate_spec
from fastapi.openapi.utils import get_openapi

app = FastAPI()

class TestEnum(enum.Enum):
    ONE = "ONE"
    #TWO = "TWO" # <-- Works whether or not you have this commented

@app.get("/items/{item_id}")
def read_item(item_id: int, q: TestEnum | None = None):
    return {"item_id": item_id, "q": q}

openapi_schema = get_openapi(
        title="Testing",
        version="0.1.0",
        routes=app.routes,
)
validate_spec(openapi_schema)
```

**Reply by dmelo** [reply]
That is a very helpful answer. Thank you.
